common-service-core/src/main/java/org/zowe/apiml/util/UrlUtils.java

```diff
@@ -112,4 +112,124 @@ public boolean isValidUrl(String urlString) {
+     */
+    public String formatHostnameForUrl(String hostname) {
+        if (hostname == null || hostname.isEmpty()) {
+            return hostname;
+        }
+
+        // If already properly formatted with brackets, return as is
+        if (hostname.startsWith("[") && hostname.contains("]")) {
+            return hostname;
+        }
+
+        // Check for hostname:port format
+        int lastColonIndex = hostname.lastIndexOf(':');
+        if (lastColonIndex > -1) {
+            String possibleHost = hostname.substring(0, lastColonIndex);
+            String possiblePort = hostname.substring(lastColonIndex + 1);
+
+            // Check if what follows the last colon is a valid port number
+            if (isValidPort(possiblePort)) {
+                // If we have a port, check if the host part is IPv6
+                if (isIPv6Address(possibleHost)) {
+                    return "[" + possibleHost + "]:" + possiblePort;
+                }
+                // If the full string is NOT IPv6, return as-is
+                if (!isIPv6Address(hostname)) {
+                    return hostname; // Regular hostname:port or IPv4:port
+                }
+                // Otherwise, fall through to check if full hostname is IPv6
+            }
+        }
+
+        // No port number, check if it's a plain IPv6 address
+        if (isIPv6Address(hostname)) {
+            return "[" + hostname + "]";
+        }
+
+        return hostname;
+    }
+
+    /**
+     * Creates a proper URL string with scheme, hostname, and port,
+     * handling IPv6 addresses correctly.
+     *
+     * @param scheme The URL scheme (http, https, etc.)
+     * @param hostname The hostname or IP address
+     * @param port The port number
+     * @return A properly formatted URL string with IPv6 address handling
+     */
+    public String getUrl(String scheme, String hostname, int port) {
+        String formattedHostname = formatHostnameForUrl(hostname);
+        return String.format("%s://%s:%d", scheme, formattedHostname, port);
+    }
+
+    /**
+     * Creates a proper URL string with scheme and host (which may include port),
+     * handling IPv6 addresses correctly.
+     *
+     * @param scheme The URL scheme (http, https, etc.)
+     * @param hostWithPort The hostname or IP address, possibly including a port
+     * @return A properly formatted URL string with IPv6 address handling
+     */
+    public String getUrl(String scheme, String hostWithPort) {
+        if (scheme == null || scheme.isEmpty()) {
+        throw new IllegalArgumentException("Scheme cannot be null or empty");
+        }
+
+        if (hostWithPort == null || hostWithPort.isEmpty()) {
+        throw new IllegalArgumentException("Host cannot be null or empty");
+        }   
+
+        // Remove any existing scheme if present
+        String cleanHostWithPort = hostWithPort.replaceFirst("^\\w+://", "");
```

Comment:
```suggestion
        String cleanHostWithPort = hostWithPort.replaceFirst("^[a-zA-Z][a-zA-Z0-9+.-]*://", "");
```
is probably more accurate according to the RFC 3986

Reply:
Thank You, I have applied your suggestion.